### PR TITLE
Added BlockPuzzle

### PR DESCRIPTION
**Pull request by konstantinosfragkoulis**
# Your checklist for this pull request

## Author name

**Author:** Konstantinos Fragkoulis

## About your game

**What is your game about?** Reaching the portal by pushing blocks, avoiding the enemy, and not running out of moves.

**How do you play your game?** Use WASD to move. If you run out of moves or the enemy catches up to you the level restarts. Push the boxes strategically to create space for you to reach the red portal.

## Code
Check off the items that are true. 
- [x] The game was made using the [Sprig editor](https://sprig.hackclub.com/editor).
- [x] The game is placed in the in the [`/games` directory](https://github.com/hackclub/sprig/tree/main/games).
- [x] The code is significantly different from all other games in the [Sprig gallery](https://sprig.hackclub.com/gallery) (except for games labeled "demo").
- [x] The game runs without errors. 
- [x] The name of the file/game contains only alphanumeric characters, `-`s, or `_`s.
- [x] The game name is not the same as the others from [gallery](https://sprig.hackclub.com/gallery)

## Image (If an image is used)

- [x] The image is in the [`/games/img` directory](https://github.com/hackclub/sprig/tree/main/games/img).
- [x] The name of the image matches the name of your file.

### CODE REVIEW

File: games/BlockPuzzle.js [hunk 1]
```diff
@@ -0,0 +1,478 @@
...
+addText((maxMoves[level]-curMoves).toString(), { 
+  x: 1,
+  y: 1,
+  color: color`3`
+})
+
+
+
+
+function findShortestPath(x, y) {
```
Comment (grymmy) [inline games/BlockPuzzle.js:299]: Excellent work!

Comment (grymmy) [reply games/BlockPuzzle.js:299]: Hmmm... after playing the game though, I am not sure if this functions as expected...  would that surprise you?  do you expect the enemy to follow the player w/ an optimal path?

**Comment by grymmy** [comment]
While I believe this game has a bug or two in it, it is playable, and obviously has had some work done on it.  I'd challenge you, @konstantinosfragkoulis, to update the game to add some polish, perhaps fixing what seems to be an issue in your pathfinding logic.

If you're a teen, please fill out [this form](https://sprig-order.hackclub.dev/) to receive your Sprig device.  Also, feel free to check out Hack Club's #sprig channel and share your game w/ the rest of the community!

**Comment by konstantinosfragkoulis** [comment]
Thanks! When I patch the issues I will submit a new pr.

**Pull request merged**